host/xtest/regression_4000.c

```diff
@@ -4186,6 +4211,11 @@ static void xtest_tee_test_4006(ADBG_Case_t *c,
 		 * the payload.
 		 */
 		if (tv->mode == TEE_MODE_VERIFY || tv->mode == TEE_MODE_SIGN) {
+			/*
+			 * Current ECDSA tests are based on SHA1 hashed payload.
+			 * Current RSASSA_PKCS1_V1_5 tests are based on SHA256
+			 * hashed payload.
+			 */
 			if (TEE_ALG_GET_MAIN_ALG(tv->algo) == TEE_MAIN_ALGO_ECDSA)
 				hash_algo = TEE_ALG_SHA1;
```

Comment:
How about deriving this from the algorithm in the case of TEE_ALG_ECDSA_SHA*?

Reply:
I have the change ready to `__tee_alg_get_digest_hash()` to support ECDSA algo IDs. I create a P-R.

Reply:
Actually, such change requires 1-to-1 relation ship between `TEE_ALG_ECDSA_P*` and `TEE_ALG_ECDSA_SHA*` is cut. That what's proposed in https://github.com/OP-TEE/optee_os/pull/7251. I'll append a commit on that P-R.